web/modules/custom/custom_weather/custom_weather.module

```diff
@@ -0,0 +1,22 @@
+<?php
+
+function custom_weather_block_info():array {
+  $blocks['custom_weather_block'] = [
+    'info' => t('Custom Weather Block'),
+    ];
+  return $blocks;
+}
+
+/**
+ * Implement hook_theme().
+ */
+function custom_weather_block_theme(): array {
+  return [
+    'custom_weather_block_template' => [
```

Comment:
Could've described what this template is for.
And "template" in the name of the template seems unnecessary.